spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteReachableFilesSparkAction.java

```diff
@@ -103,53 +87,46 @@ public DeleteReachableFiles io(FileIO fileIO) {
 
   @Override
   public DeleteReachableFiles deleteWith(Consumer<String> deleteFunc) {
-    this.removeFunc = deleteFunc;
+    this.deleteFunc = deleteFunc;
     return this;
-
   }
 
   @Override
   public DeleteReachableFiles executeDeleteWith(ExecutorService executorService) {
-    this.removeExecutorService = executorService;
+    this.deleteExecutorService = executorService;
     return this;
   }
 
   @Override
   public Result execute() {
-    Preconditions.checkArgument(io != null, "File IO cannot be null");
-    String msg = String.format("Removing files reachable from %s", tableMetadata.metadataFileLocation());
-    JobGroupInfo info = newJobGroupInfo("REMOVE-FILES", msg);
+    String jobDesc = String.format("Deleting files reachable from %s", metadataFileLocation);
+    JobGroupInfo info = newJobGroupInfo("DELETE-REACHABLE-FILES", jobDesc);
     return withJobGroupInfo(info, this::doExecute);
   }
 
   private Result doExecute() {
-    boolean streamResults = PropertyUtil.propertyAsBoolean(options(), STREAM_RESULTS, false);
-    Dataset<Row> validFileDF = buildValidFileDF(tableMetadata).distinct();
+    TableMetadata metadata = TableMetadataParser.read(io, metadataFileLocation);
+
+    ValidationException.check(
+        PropertyUtil.propertyAsBoolean(metadata.properties(), GC_ENABLED, GC_ENABLED_DEFAULT),
+        "Cannot delete files: GC is disabled (deleting files may corrupt other tables)");
+
+    Dataset<Row> reachableFileDF = buildReachableFileDF(metadata).distinct();
+
+    boolean streamResults = PropertyUtil.propertyAsBoolean(options(), STREAM_RESULTS, STREAM_RESULTS_DEFAULT);
     if (streamResults) {
-      return deleteFiles(validFileDF.toLocalIterator());
+      return deleteFiles(reachableFileDF.toLocalIterator());
     } else {
-      return deleteFiles(validFileDF.collectAsList().iterator());
+      return deleteFiles(reachableFileDF.collectAsList().iterator());
     }
   }
 
-  private Dataset<Row> projectFilePathWithType(Dataset<Row> ds, String type) {
-    return ds.select(functions.col("file_path"), functions.lit(type).as("file_type"));
-  }
-
-  private Dataset<Row> buildValidFileDF(TableMetadata metadata) {
+  private Dataset<Row> buildReachableFileDF(TableMetadata metadata) {
     Table staticTable = newStaticTable(metadata, io);
-    return projectFilePathWithType(buildValidContentFileDF(staticTable), CONTENT_FILE)
-        .union(projectFilePathWithType(buildManifestFileDF(staticTable), MANIFEST))
-        .union(projectFilePathWithType(buildManifestListDF(staticTable), MANIFEST_LIST))
-        .union(projectFilePathWithType(buildOtherMetadataFileDF(staticTable), OTHERS));
-  }
-
-  @Override
-  protected Dataset<Row> buildOtherMetadataFileDF(Table table) {
```

Comment:
This used to overload the method in the base class. I defined `buildAllReachableOtherMetadataFileDF` instead.